hive-hooks/src/main/java/com/expediagroup/apiary/extensions/hooks/filters/ApiaryMetastoreFilter.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (C) 2018-2021 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.apiary.extensions.hooks.filters;
+
+import java.util.List;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.MetaStoreFilterHook;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.Index;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.PartitionSpec;
+import org.apache.hadoop.hive.metastore.api.Table;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.expediagroup.apiary.extensions.hooks.config.Configuration;
+import com.expediagroup.apiary.extensions.hooks.converters.GenericConverter;
+import com.expediagroup.apiary.extensions.hooks.pathconversion.config.PathConversionConfiguration;
+import com.expediagroup.apiary.extensions.hooks.pathconversion.converters.PathConverter;
+
+public class ApiaryMetastoreFilter implements MetaStoreFilterHook {
+
+  private final Configuration configuration;
+  private final GenericConverter converter;
+
+  public ApiaryMetastoreFilter(HiveConf conf) {
+    configuration = new PathConversionConfiguration(conf);
+    converter = new PathConverter(configuration);
```

Comment:
```suggestion
    this(conf, new PathConverter(configuration));
```

Reply:
I'm not 100% sure I understand this change suggestion. Is the idea to change the default constructor to just only be referencing the test code?
```
  public ApiaryMetastoreFilter(HiveConf conf) {
    this(conf, new PathConverter(new PathConversionConfiguration(conf)));
  }

  @VisibleForTesting
  ApiaryMetastoreFilter(HiveConf conf, GenericConverter schemeConverter) {
    configuration = new PathConversionConfiguration(conf);
    converter = schemeConverter;
  }
```
In this case we'd be duplicating the configuration definition, which doesn't seem ideal. Or is the idea to more so refactor this approach for `PathConverter` to accept the plain `HiveConf conf` var and create the specific configuration at the GenericConverter level instead of the ApiaryMetastoreFilter level?

Reply:
Yes, it was so the one constructor could delegate to the other to remove duplicated code but I hadn't spotted that you actually need to do `new PathConverter(new PathConversionConfiguration(conf))` which duplicates it again. I guess best to leave it as you have it then.